### PR TITLE
DEX-1199 Default settings for checker, refactors

### PR DESCRIPTION
**Pull request by alexander-branevskiy**

### CODE REVIEW

File: dex/src/main/resources/application.conf [hunk 1]
```diff
@@ -14,7 +14,8 @@ waves.dex {
   # It is recommended to change to matcher-1, matcher-2, ... in a cluster setup.
   id = "default"
 
-  address-scheme-character = ""
+  # T - testnet, D - devnet, S - stagenet
```
Comment (vsuharnikov) [inline dex/src/main/resources/application.conf:17]: There is MainNet:
```
# W - mainnet, T - testnet, S - stagenet, D - devnet
```

File: dex/src/main/scala/com/wavesplatform/dex/cli/WavesDexCli.scala [hunk 2]
```diff
@@ -1,43 +1,43 @@
...
+import java.util.{Base64, Scanner}
 import scala.concurrent.duration.{DurationInt, FiniteDuration}
+import scala.concurrent.{Await, TimeoutException}
 import scala.util.{Failure, Success, Try, Using}
 
 object WavesDexCli extends ScoptImplicits {
 
   private val backend = HttpURLConnectionBackend()
 
+  // noinspection ScalaStyle
```
Comment (vsuharnikov) [inline dex/src/main/scala/com/wavesplatform/dex/cli/WavesDexCli.scala:40]: Wondering, why do we need this?

File: dex/src/main/scala/com/wavesplatform/dex/cli/WavesDexCli.scala [hunk 3]
```diff
@@ -456,28 +448,30 @@ object WavesDexCli extends ScoptImplicits {
...
-            case Command.CheckServer => checkServer(args)
-            case Command.RunComparison => runComparison(args)
-            case Command.MakeOrderbookSnapshots => makeSnapshots(args)
-            case Command.CheckConfigFile => checkConfig(args)
-            case Command.CleanAssets => cleanAssets(args)
+            case Command.GenerateAccountSeed => generateAccountSeed(updatedArgs)
+            case Command.CreateAccountStorage => createAccountStorage(updatedArgs)
+            case Command.CreateDocumentation => createDocumentation(updatedArgs)
+            case Command.CreateApiKey => createApiKey(updatedArgs)
+            case Command.CheckServer => checkServer(updatedArgs)
```
Comment (vsuharnikov) [inline dex/src/main/scala/com/wavesplatform/dex/cli/WavesDexCli.scala:463]: We could pass loaded matcher settings from:
```scala
val argsOverrides = loadMatcherSettingsUnsafe(args.configPath).cli.argsOverrides
```

File: dex/src/main/scala/com/wavesplatform/dex/cli/WavesDexCli.scala [hunk 4]
```diff
@@ -141,9 +140,9 @@ object WavesDexCli extends ScoptImplicits {
              |  Account seed          : ${args.accountSeed.getOrElse("")}
                    """.stripMargin
         )
-        (config, matcherConfig) <- cli.wrapByLogs("  Loading Matcher settings... ")(loadAllConfigs(args.configPath))
-        superConnector <- SuperConnector.create(matcherConfig, apiUrl, args.nodeRestApi, args.authServiceRestApi, apiKey)
-        checkResult <- new Checker(superConnector).checkState(args.version, args.accountSeed, config, matcherConfig)
+        (config, matcherSettings) <- cli.wrapByLogs("  Loading Matcher settings... ")(loadAllConfigs(args.configPath))
```
Comment (vsuharnikov) [inline dex/src/main/scala/com/wavesplatform/dex/cli/WavesDexCli.scala:143]: It looks like we ca move `matcherSettings` to the `checkServer` parameters.

**Review by vsuharnikov — CHANGES_REQUESTED** [review]

File: dex/src/main/scala/com/wavesplatform/dex/cli/WavesDexCli.scala (hunk 2 above)
Comment (alexander-branevskiy) [reply dex/src/main/scala/com/wavesplatform/dex/cli/WavesDexCli.scala:40]: the function below uses ```println``` which will be falsely highlighted by Idea

**Review by vsuharnikov — APPROVED** [review]

File: dex/src/main/scala/com/wavesplatform/dex/cli/WavesDexCli.scala (hunk 3 above)
Comment (alexander-branevskiy) [reply dex/src/main/scala/com/wavesplatform/dex/cli/WavesDexCli.scala]: resolved

File: dex/src/main/scala/com/wavesplatform/dex/cli/WavesDexCli.scala (hunk 4 above)
Comment (alexander-branevskiy) [reply dex/src/main/scala/com/wavesplatform/dex/cli/WavesDexCli.scala]: resolved

**Pull request merged**